osiris/lib/db/mongo.go

```diff
@@ -45,20 +47,23 @@ var MongoDatabaseCliOptions = libcli.CommonCliOptions(libcli.Mongo)
 // MongoDatabase ...
 func MongoDatabase(config *MongoDBConfig) (*MongoDB, error) {
 	databaseName := config.Database
-	databaseAuth := fmt.Sprintf("%s:%s", config.User, config.Password)
+	var databaseAuth string
+	if len(config.User+config.Password) > 0 {
+		databaseAuth = fmt.Sprintf("%s:%s@", config.User, config.Password)
+	}
 	databaseHost := fmt.Sprintf("%s:%d", config.Host, config.Port)
-	databaseConnectionURI := fmt.Sprintf("mongodb://%s@%s/?connect=direct", databaseAuth, databaseHost)
+	databaseConnectionURI := fmt.Sprintf("mongodb://%s%s/?connect=direct", databaseAuth, databaseHost)
 	client, err := mongo.NewClient(options.Client().ApplyURI(databaseConnectionURI))
```

Comment:
I think it's nicer to keep the @ in line 55. After all it does not really belong to the database auth itself :)

Reply:
Nope. Only @ is invalid